train_segmentation.py

```diff
@@ -403,65 +483,167 @@ def main(params, config_path):
+    overlap = get_key_def("overlap", params["sample"], default=5, expected_type=int)
+    min_annot_perc = get_key_def('min_annotated_percent', params['sample']['sampling_method'], default=0,
+                                 expected_type=int)
+    if not data_path.is_dir():
+        raise FileNotFoundError(f'Could not locate data path {data_path}')
     samples_folder_name = (f'samples{samples_size}_overlap{overlap}_min-annot{min_annot_perc}_{num_bands}bands'
                            f'_{experiment_name}')
     samples_folder = data_path.joinpath(samples_folder_name)
-    batch_size = params['training']['batch_size']
-    num_devices = params['global']['num_gpus']
-    # # list of GPU devices that are available and unused. If no GPUs, returns empty list
-    max_used_ram = get_key_def('max_used_ram', params['global'], 2000, expected_type=int)
-    max_used_perc = get_key_def('max_used_perc', params['global'], 15, expected_type=int)
-    lst_device_ids = get_device_ids(
-        num_devices, max_used_ram=max_used_ram, max_used_perc=max_used_perc, debug=debug) \
-        if torch.cuda.is_available() else []
-    num_devices = len(lst_device_ids) if lst_device_ids else 0
-    device = torch.device(f'cuda:{lst_device_ids[0]}' if torch.cuda.is_available() and lst_device_ids else 'cpu')
-
-    tqdm.write(f'Creating dataloaders from data in {samples_folder}...\n')
+
+    # visualization parameters
+    vis_at_train = get_key_def('vis_at_train', params['visualization'], default=False)
+    vis_at_eval = get_key_def('vis_at_evaluation', params['visualization'], default=False)
+    vis_batch_range = get_key_def('vis_batch_range', params['visualization'], default=None)
+    vis_at_checkpoint = get_key_def('vis_at_checkpoint', params['visualization'], default=False)
+    ep_vis_min_thresh = get_key_def('vis_at_ckpt_min_ep_diff', params['visualization'], default=1, expected_type=int)
+    vis_at_ckpt_dataset = get_key_def('vis_at_ckpt_dataset', params['visualization'], 'val')
+
+    # coordconv parameters
+    coordconv_params = {}
+    for param, val in params['global'].items():
+        if 'coordconv' in param:
+            coordconv_params[param] = val
+
+    # add git hash from current commit to parameters if available. Parameters will be saved to model's .pth.tar
+    params['global']['git_hash'] = get_git_hash()
+
+    # automatic model naming with unique id for each training
+    model_id = config_path.stem
+    output_path = samples_folder.joinpath('model') / model_id
+    if output_path.is_dir():
+        last_mod_time_suffix = datetime.fromtimestamp(output_path.stat().st_mtime).strftime('%Y%m%d-%H%M%S')
+        archive_output_path = samples_folder.joinpath('model') / f"{model_id}_{last_mod_time_suffix}"
+        shutil.move(output_path, archive_output_path)
+    output_path.mkdir(parents=True, exist_ok=False)
+    shutil.copy(str(config_path), str(output_path))  # copy yaml to output path where model will be saved
```

Comment:
@victorlazio109 see how this is different than images_to_samples.py. No exceptions raised.

Reply:
seen, thanks